src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -92,14 +94,18 @@ private void Run()
             {
                 _delayStart.TrySetException(ex);
             }
+            _serviceStopped.TrySetResult(null);
```

Comment:
Your other question about exceptions makes me wonder who would ever observe an exception thrown from this.  I think we could also propagate an exception if one occurred during stop - though it would likely be our own, since the only thing running on the dispatcher thread is ServiceBase.

Reply:
>  I think we could also propagate an exception if one occurred during stop - though it would likely be our own, since the only thing running on the dispatcher thread is ServiceBase.

That's another answer to the other question: if we catch all exceptions inside of Stop and use one to complete _serviceStopped, then the code in StopAsync can remain as is, as Stop itself would never fail and would instead propagate its exception to _serviceStopped.

Reply:
I think I'll do this for correctness.  There's no reason to hide exceptions from framework code -- even if we don't expect them.  That just makes things harder to debug in the future.